### ISSUE TITLE
data-sort-value not working as expected

### ISSUE THREAD

**Issue by fitzage** [fooplugins/FooTable#549]
I have a table where one column lists prices along the lines of `$0.060` and if the price is zero, it lists `Free`. So for sorting, I have `data-sort-value='0'` for the free ones, and `data-sort-value='0.060'` for the ones with a price.

In FooTable 2, I also set the column type as numeric, and this sorted correctly. In the current version, if I set the column type to `number`, it removes all the `Free` text, and displays those cells as blank (although it does sort in the correct order). If I don't set the column type (or set it to html), it seems to completely ignore my sort value, because there ends up being no apparent rhyme or reason to how the column gets sorted. Most of the free items end up on top, and then a couple of items with prices, then more free items, then the rest of the items with prices, and then more free items. Perhaps this is because alphabetic sort doesn't really handle numbers of this format when sorting?

At any rate, if I could set it to "number" without deleting what I put in the table cells in the first place, that would solve the problem. Is there a setting for that?

**Comment by steveush** [comment]
Hi,

I would use the [column formatter option](http://fooplugins.github.io/FooTable/docs/getting-started.html#column-formatter) and set the column type to `number`. In the formatter function perform a check on the given value and output what I wanted to see in the cell.

e.g.

``` html
<script>
    function priceFormat(value){
        return value === 0 ? "Free" : "$ " + value;
    }
</script>
<table>
    <thead>
        <tr>
            <th data-type="number" data-formatter="priceFormat">Price</th>
            ...
        </tr>
    </thead>
    <tbody>
        <tr>
            <td>0</td>
            ...
        </tr>
        <tr>
            <td>0.060</td>
            ...
        </tr>
    </tbody>
</table>
```

This way I wouldn't have to set the `data-sort-value` attributes or pre-format the cell contents.

The behavior you saw when setting the column type to html is unexpected though and I could not replicate it my side. If you set the type to `html` and then supply a `data-sort-value` for each cell the sorting component should be using those values to sort. If you leave the `data-sort-value` attribute off the cells of a `html` column you might see strange sorting behavior as the cell is sorted using it's contents, which is HTML.

**Comment by fitzage** [comment]
Excellent, thanks. Not sure what's going on with the sorting when I set it to html, but I'll try this method.

**Comment by fitzage** [comment]
OK, this works perfectly. The downside is that if someone has javascript disabled, they'll just see numbers not the pretty formatting.

**Comment by steveush** [comment]
@fitzage Which version of FooTable are you using by the way? There's been a second HTML sorting related issue logged and I'm trying to track down what the issue is.

Thanks

**Comment by steveush** [comment]
@fitzage The HTML sorting issue is fixed in 3.1.3 which should allow you to perform sorting as you originally tried if you prefer to use that approach.

Thanks

**Comment by fitzage** [comment]
Rock on. Thanks. Fixed before I even had a chance to respond to the previous comments.
